lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -170,4 +171,81 @@ module.exports = function(
     });
   };
 
+  /**
+   * @summary Memoize ImmutableJS list reference
+   * @function
+   * @private
+   *
+   * @description
+   * This workaround is needed to avoid AngularJS from getting
+   * caught in an infinite digest loop when using `ngRepeat`
+   * over a function that returns a mutable version of an
+   * ImmutableJS object.
+   *
+   * The problem is that every time you call `myImmutableObject.toJS()`
+   * you will get a new object, whose reference is different from
+   * the one you previously got, even if the data is exactly the same.
+   *
+   * @param {Function} func - function that returns an ImmutableJS list
+   * @param {Function} compare - function used to compare the old and new lists
+   * @returns {Function} memoized function
+   *
+   * @example
+   * const getList = () => {
+   *   return Store.getState().toJS().myList;
+   * };
+   *
+   * const memoizedFunction = memoizeImmutableListReference(getList);
+   */
+  const memoizeImmutableListReference = (func, compare = _.isEqual) => {
+    let previous = [];
+
+    return (...restArgs) => {
+      const list = Reflect.apply(func, this, restArgs);
+
+      if (!_.isEqualWith(list, previous, compare)) {
+        previous = list;
+      }
+
+      return previous;
+    };
+  };
+
+  /**
+   * @summary Get the list of drives and their compatibility status objects
+   * @function
+   * @public
+   *
+   * @description
+   * For each drive, return its compatibility status with the seleted image,
+   * containing the status type (ERROR, WARNING), and accompanying
+   * status message.
+   *
+   * @returns {Object[]} list of objects containing drive and statuses
+   *
+   * @example
+   * const metadrives = DriveSelectorController.getDrivesAndStatuses();
+   *
+   * for ({ drive, statuses } of metadrives) {
+   *   for ({ type, message } of statuses) {
+   *     // do something
+   *   }
+   * }
+   */
+  this.getDrivesAndStatuses = memoizeImmutableListReference(() => {
+    const statusList = _.map(this.drives.getDrives(), (drive) => {
+      return {
+        drive,
+        statuses: this.constraints.getDriveImageCompatibilityStatuses(drive, this.state.getImage())
+      };
+    });
+
+    return statusList;
+  }, (newList, oldList) => {
+    return _.isEqual(
+      utils.removeAngularHashes(newList),
```

Comment:
Awesome find of `angular.equals`! Unfortunately this doesn't wholly work because `memoizeImmutableListReference` would need to memoize on a per-argument basis, which is messy to do in a generic manner—when it's called with a different argument, the previous argument's returned list is used. So we can definitely get rid of the `removeAngularHashes` utility function, but not `getDrivesAndStatuses` (yet anyway).